packages/website/ts/components/ui/icon_button.tsx

```diff
@@ -0,0 +1,63 @@
+import { colors, Styles } from '@0xproject/react-shared';
+import * as _ from 'lodash';
+import * as React from 'react';
+
+export interface IconButtonProps {
+    iconName: string;
+    labelText?: string;
+    onClick: () => void;
+    color?: string;
+}
+interface IconButtonState {
+    isHovering: boolean;
+}
+export class IconButton extends React.Component<IconButtonProps, IconButtonState> {
+    public static defaultProps: Partial<IconButtonProps> = {
+        onClick: _.noop,
+        labelText: '',
+        color: colors.mediumBlue,
+    };
+    public constructor(props: IconButtonProps) {
+        super(props);
+        this.state = {
+            isHovering: false,
+        };
```

Comment:
I think you can make this component stateless and just user the css :hover selector for this hover effect. https://www.w3schools.com/cssref/sel_hover.asp

We do have this all.less file where we can write less. However this also works :)

Reply:
Ok I agree with that. I do think that this type of thing is better suited for CSS but I agree that having the styles co-located with the component code trumps that. 

*Thoughts*
However, I do think we will run into this more often, like, we need to be able to use CSS. Psuedo selectors and media queries are going to come up -- also it's just generally not great that we're using inline styles so heavily, although basscss and material UI is doing a lot of the heavy lifting. We could use css-modules, but we would need to change the structure of our project (one directory per component so that the stylesheets can live next to the component), or we can use CSS-IN-JS like styled-components. But yea not relevant now --- but in the future 😄